pylint/config/callback_actions.py

```diff
@@ -38,3 +47,261 @@ def __call__(
         option_string: Optional[str] = None,
     ) -> None:
         return None
+
+
+class _AccessRunObjectAction(_CallbackAction):
+    """Action that has access to the Run object."""
+
+    def __init__(
+        self,
+        option_strings: Sequence[str],
+        dest: str,
+        nargs: None = None,
+        const: None = None,
+        default: None = None,
+        arg_type: None = None,
+        choices: None = None,
+        required: bool = False,
+        help_msg: str = "",
+        metavar: str = "",
+        **kwargs: "Run",
+    ) -> None:
+        self.run = kwargs["Run"]
+
+        super().__init__(
+            option_strings,
+            dest,
+            0,
+            const,
+            default,
+            arg_type,
+            choices,
+            required,
+            help_msg,
+            metavar,
+        )
+
+    @abc.abstractmethod
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = None,
+    ) -> None:
+        raise NotImplementedError
+
+
+class _MessageHelpAction(_CallbackAction):
+    """Display the help message of a message."""
+
+    def __init__(
+        self,
+        option_strings: Sequence[str],
+        dest: str,
+        nargs: None = None,
+        const: None = None,
+        default: None = None,
+        arg_type: None = None,
+        choices: None = None,
+        required: bool = False,
+        help_msg: str = "",
+        metavar: str = "",
+        **kwargs: "Run",
+    ) -> None:
+        self.run = kwargs["Run"]
+        super().__init__(
+            option_strings,
+            dest,
+            "+",
+            const,
+            default,
+            arg_type,
+            choices,
+            required,
+            help_msg,
+            metavar,
+        )
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[str], None],
+        option_string: Optional[str] = "--help-msg",
+    ) -> None:
+        if isinstance(values, (list, tuple)):
+            self.run.linter.msgs_store.help_message(values)
+        elif isinstance(values, str):
+            self.run.linter.msgs_store.help_message(utils._splitstrip(values))
+        else:
+            raise argparse.ArgumentTypeError(f"{values} should be a string.")
```

Comment:
```suggestion
```
Isn't all arguments from argparse string or list ? We could add a test for this otherwise.

Reply:
Removed and changed tests to "catch" this case. I doesn't go here as `argparse` errors out earlier when the `values` is `None`.